core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1922,6 +1922,18 @@ public SqlNode field(int ordinal, boolean useAlias) {
           needNew && !aliases.containsKey(neededAlias) ? newAliases : aliases);
     }
 
+    private boolean isAliasRefNeeded(boolean keepColumnAlias, RelNode rel) {
+      if (expectedClauses.contains(Clause.HAVING)
```

Comment:
can we combine these condition to make it more readable 
'if ((expectedClauses.contains(Clause.HAVING) && dialect.getConformance().isHavingAlias()) || keepColumnAlias) {
        return true;
    }'